File: Java/345.java
__________________________________________________________________________________________________
sample 1 ms submission
import java.util.*;
import static java.util.Arrays.asList;

class Solution {
    
    public String reverseVowels(String s) {
        if( s == null || s.length() == 0 ) return s;
        boolean[] vowels = new boolean[256];
        char[] v = { 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' };
        for( char c : v ) vowels[c] = true;
        int left = 0, right = s.length() - 1;
        char[] array = s.toCharArray();
        while( left < right ) {
            while( left < right && !vowels[array[left]] ) left++;
            while( right > left && !vowels[array[right]] ) right--;
            if( left >= right ) break;
            char temp = array[left];
            array[left] = array[right];
            array[right] = temp;
            left++;
            right--;
        }
        return new String( array );
    }
}
__________________________________________________________________________________________________
sample 35940 kb submission
class Solution {
    public String reverseVowels(String s) {
      //input validation
        if(s==null || s.length()==0)
            return s;
          char[] result = s.toCharArray();  // is a must to let swap works
        int start= 0 , end = s.length()-1;
       while( start < end){
           
           while (start < end && !isVowel(result[start]) )  // start < end should be added not to exceed the string's length
               start++;
           
            while (start < end && !isVowel(result[end]))
              end--;
           
           swap(result, start , end );
           
           start++;  // for new start and end
            end--;
           
       }
   // return s; // dont forget return statment.
        return new String(result);
    }  
    
    private boolean isVowel(char c ){
        c = Character.toLowerCase(c);
        if (c == 'a' ||c == 'e' || c == 'o' || c == 'i' || c == 'u' )
            return true;
        else return false;
    }
    
    private void swap(char[] s , int i , int j){
        if (s[i] == s[j]) return; // no need for swap
        
         char temp =  s[i];
         s[i] = s[j];
         s[j] = temp;
    }    
}
__________________________________________________________________________________________________

File: Python3/345.py
__________________________________________________________________________________________________
sample 36 ms submission
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        l = len(s)
        p1 = 0
        p2 = l - 1
        f1 = f2 = False
        set_ = {'a', 'e', 'i', 'o', 'u','A','E','I','O','U'}
        while p1 < p2:
            if s[p1] in set_:
                while (p1 < p2):
                    if s[p2] in set_:
                        tem = s[p1]
                        s[p1] = s[p2]
                        s[p2] = tem
                        p1+=1
                        p2-=1
                        break
                    else:
                        p2 -= 1
            else:
                p1 += 1
        return ''.join(s)
__________________________________________________________________________________________________
sample 13768 kb submission
class Solution:
    def reverseVowels(self, s: str) -> str:
        
        ss = ['a','e','i','o','u','A','E','I','O','U']
        temp = []
        for i in s:
            if i in ss:
                temp.append(i)
        res = ''
        for i in s:
            if i in ss:
                
                res+=temp.pop()
            else:
                res+=i 

        return res
__________________________________________________________________________________________________
